common/lib/xmodule/xmodule/js/src/word_cloud/word_cloud_main.js

```diff
@@ -154,136 +154,166 @@
+                    '{listStart}{startTag}{word}{endTag}{percent}{listEnd}',
+                    {
+                        listStart: HtmlUtils.HTML('<li>'),
+                        startTag: HtmlUtils.HTML('<strong>'),
+                        word: word,
+                        endTag: HtmlUtils.HTML('</strong>'),
+                        percent: percent,
+                        listEnd: HtmlUtils.HTML('</li>')
+                    }
+                ).toString());
             });
-            studentWordsStr = '' + studentWordsKeys.join(', ');
+
+            studentWordsStr = '' + studentWordsKeys.join('');
 
             cloudSectionEl
-            .addClass('active')
-            .find('.your_words').html(studentWordsStr)
-            .end()
-            .find('.total_num_words').html(response.total_count);
+                .addClass('active');
+
+            HtmlUtils.setHtml(
+                cloudSectionEl.find('.your_words'),
+                HtmlUtils.HTML(studentWordsStr)
+            );
+
+            cloudSectionEl.find('.your_words').end().find('.total_num_words')
+                .text(response.total_count);
 
             $(cloudSectionEl.attr('id') + ' .word_cloud').empty();
 
-        // Actual drawing of word cloud.
+            // Actual drawing of word cloud.
             groupEl = d3.select('#' + cloudSectionEl.attr('id') + ' .word_cloud').append('svg')
-                .attr('width', this.width)
-                .attr('height', this.height)
-                .append('g')
-                .attr('transform', 'translate(' + (0.5 * this.width) + ',' + (0.5 * this.height) + ')')
-                .selectAll('text')
-                .data(words)
-                .enter().append('g');
+                    .attr('width', this.width)
+                    .attr('height', this.height)
+                    .append('g')
+                    .attr('transform', 'translate(' + (0.5 * this.width) + ',' + (0.5 * this.height) + ')')
+                    .selectAll('text')
+                    .data(words)
+                    .enter()
+                    .append('g')
+                    .attr('aria-describedby', function(d) {
+                        return HtmlUtils.interpolateHtml(
+                            gettext('text_word_{word} title_word_{word}'),
+                            {
+                                word: d.text.replace(/\s/g, '_')
+                            }
+                        );
+                    });
 
             groupEl
-            .append('title')
-            .text(function(d) {
-                var res = '';
-
-                $.each(response.top_words, function(index, value) {
-                    if (value.text === d.text) {
-                        res = value.percent + '%';
-
-                        return;
-                    }
+                .append('title')
+                .attr('id', function(d) {
+                    return 'title_word_' + d.text;
```

Comment:
you need to replace all of the spaces in d.text with underscores if you want the ID attributes to work.